post-processor/compress/post-processor.go

```diff
@@ -2,97 +2,308 @@ package compress
 
 import (
 	"archive/tar"
+	"archive/zip"
 	"compress/gzip"
 	"fmt"
 	"io"
 	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
 
+	"github.com/klauspost/pgzip"
 	"github.com/mitchellh/packer/common"
 	"github.com/mitchellh/packer/helper/config"
 	"github.com/mitchellh/packer/packer"
 	"github.com/mitchellh/packer/template/interpolate"
+	"github.com/pierrec/lz4"
 )
 
 type Config struct {
 	common.PackerConfig `mapstructure:",squash"`
-
-	OutputPath string `mapstructure:"output"`
-
-	ctx interpolate.Context
+	OutputPath          string `mapstructure:"output"`
+	CompressionLevel    int    `mapstructure:"compression_level"`
+	KeepInputArtifact   bool   `mapstructure:"keep_input_artifact"`
+	Archive             string
+	Algorithm           string
+	UsingDefault        bool
```

Comment:
Nitpick, but can you add a newline after this? I like to separate the "categories" of options, and exported/unexported usually fits the bill.